cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java

```diff
@@ -543,6 +549,38 @@ public Boolean call() throws Exception {
     assertNoAccess(appId);
   }
 
+  @Test
+  public void testSearchMetadata() throws Exception {
+    createAuthNamespace();
+    deployApplication(AUTH_NAMESPACE.toId(), AllProgramsApp.class);
+    MetadataAdmin metadataAdmin = getMetadataAdmin();
+    Id.Stream streamId = Id.Stream.from(AUTH_NAMESPACE.toId(), AllProgramsApp.STREAM_NAME);
+
+    metadataAdmin.addProperties(streamId, ImmutableMap.of("key1", "value1"));
+    metadataAdmin.addTags(streamId, "tag1", "tag2");
+
+    Set<MetadataSearchResultRecord> result = metadataAdmin.searchMetadata(AUTH_NAMESPACE.getNamespace(),
+                                                                          "value*",
+                                                                          ImmutableSet.<MetadataSearchTargetType>of());
+    // Result is empty now since Alice don't have the privileges on it
+    Assert.assertTrue(result.isEmpty());
+    // Grant access and search again
+    grantAndAssertSuccess(streamId.toEntityId(), ALICE, ImmutableSet.of(Action.READ));
+    result = metadataAdmin.searchMetadata(AUTH_NAMESPACE.getNamespace(),
+                                          "value*",
+                                          ImmutableSet.<MetadataSearchTargetType>of());
+    Assert.assertFalse(result.isEmpty());
+
+    MetadataSearchResultRecord record = (MetadataSearchResultRecord) result.toArray()[0];
+    Assert.assertEquals(record.getEntityId(), streamId);
+    Metadata systemMetadata = record.getMetadata().get(MetadataScope.SYSTEM);
```

Comment:
if you just create an expected MetadataSearchResultRecord, and assertEquals the entire record on line 575, you don't need any of these checks.